### PR TITLE
Topic/amazonka develop

### PR DESCRIPTION
**Pull request by EduardSergeev**
Use latest `amazonka` (master/develop branch) which among other things has more convenient exception handling.

### CODE REVIEW

**Comment by EduardSergeev** [comment]
After today's discussion with @markhibberd and @charleso I actually very much like @etorreborre approach. I guess it is better to be decided on case by case basis how to express possible error outcome (with `Maybe`, `Either` or `EitherT` and what error type to use in `EitherT`). But it seems that the boilerplate code can be indeed eliminated in user code with function like `get`, which is internally just:
```haskell
get :: (MonadThrow m, Exception e) => EitherT e m a -> m a
get = eitherT throwM return
```
So the consumer of the function can decided for himself if he wants to handle the possible error case explicitly or just to `throwM` it and let global exception handler to handle it later. Error type need to be `Exception` for it to work but this is about it.

File: mismi-sqs/test/Test/Mismi/SQS.hs [hunk 1]
```diff
@@ -55,9 +57,10 @@ runSQSWithQueueArg v (Queue qn r) f =
 
 runSQSWithCfgWithDefaults :: AWS b -> IO b
 runSQSWithCfgWithDefaults f = do
-  r <- getRegionFromEnv
-  let e = fromMaybe Sydney $ P.rightToMaybe r
-  runSQSWithRegion e f
+  r <- getRegionFromEnv `catch` \case
+     RegionMissing -> return Sydney
```
Comment (charleso) [inline mismi-sqs/test/Test/Mismi/SQS.hs]: This is definitely one of those cases where `catch` is being used as control flow - `getRegionFromEnv` should just return `Maybe Region` or thereabouts

Comment (EduardSergeev) [inline mismi-sqs/test/Test/Mismi/SQS.hs]: Indeed it is. Will change to use `Maybe` (as `lookupEnv` does).

**Comment by charleso** [comment]
We are entering a new era. :)

**Comment by domdere** [comment]
We are Indeed.

**Comment by tranma** [comment]
image

**Comment by nhibberd** [comment]
:fireworks:

**Pull request merged**